docs/output.md

```diff
@@ -11,6 +11,13 @@ This document describes the output produced by the pipeline. Most of the plots a
 The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.
 
 <!-- TODO nf-core: Write this documentation describing your workflow's output -->
+Reading this you likely have already experienced first hand that single-cell sequencing data is big, exceeding the amount of RAM available to most computers. Nonetheless, algorithms may demand expression values of a particular set of genes for a particular set of cells at any time.
+The [hiearchical data format 5](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) (hdf5) thus provides the data in a structured manner with indices such that computational demands are mnimized to retrieve individual records. And much like a traditional file system, or like an excel sheet with many named tabs, different pieces of information can be integrated into a single consistent ..AnnData object that is effectively mapped to a file on the hard drive.
+
+Various steps in the pipeline produce outputs that need to be added to .obs or obsm etc. If the pipeline inputs are really large (10s of GB), always storing the whole AnnData object with the new info and in the end extracting only the relevant info from all objects will use a lot of disk space. To prevent this, this pipeline only stores the delta in each step and finally aggregates all deltas onto a single anndata object.
+
+Each output is put into a directory named like the prefix (until first underline) of the process creating it.
+This is still the default of nf-core, but can and should be changed, but has so far not been relevant for functionality development.
```

Comment:
This was more a message to @smoe, not something that should make it into the actual documentation :D

Reply:
What I meant is that as part of this PR it would make sense to also rearrange the outputs. Naming the directories with the process prefixes often leads to collisions and people then have a hard time keeping them apart.